bgpd/bgp_snmp.c

```diff
@@ -29,10 +29,101 @@
 #include "bgpd/bgp_snmp_bgp4.h"
 #include "bgpd/bgp_snmp_bgp4v2.h"
 #include "bgpd/bgp_mplsvpn_snmp.h"
+#include "bgpd/bgp_snmp_clippy.c"
+
+uint32_t bgp_snmp_traps_flags;
+
+static int bgp_cli_snmp_traps_config_write(struct vty *vty);
+
+DEFPY (bgp_snmp_traps_rfc4273,
+       bgp_snmp_traps_rfc4273_cmd,
+       "bgp snmp traps rfc4273<enable$on|disable$off>",
+       BGP_STR
+       "Configure BGP SNMP\n"
+       "Configure SNMP traps for BGP\n"
+       "Configure use of rfc4273 SNMP traps for BGP\n"
+       "Enable rfc4273 traps\n"
+       "Disable rfc4273 traps\n")
+{
+	if (on)
+		SET_FLAG(bgp_snmp_traps_flags,
+				BGP_SNMP_TRAPS_RFC4273_ENABLED);
+	else
+		UNSET_FLAG(bgp_snmp_traps_flags,
+			   BGP_SNMP_TRAPS_RFC4273_ENABLED);
+	return CMD_SUCCESS;
+}
+
+DEFPY (bgp_snmp_traps_bgp4_mibv2,
+       bgp_snmp_traps_bgp4_mibv2_cmd,
+       "bgp snmp traps bgp4-mibv2<enable$on|disable$off>",
+       BGP_STR
+       "Configure BGP SNMP\n"
+       "Configure SNMP traps for BGP\n"
+       "Configure use of BGP4-MIBv2 SNMP traps for BGP\n"
+       "Enable BGP4-MIBv2 traps\n"
+       "Disable BGP4-MIBv2 traps\n")
+{
+	if (on)
+		SET_FLAG(bgp_snmp_traps_flags,
+				BGP_SNMP_TRAPS_BGP4MIBV2_ENABLED);
+	else
+		UNSET_FLAG(bgp_snmp_traps_flags,
+			   BGP_SNMP_TRAPS_BGP4MIBV2_ENABLED);
+	return CMD_SUCCESS;
+}
+
+static void bgp_snmp_traps_init(void)
+{
+	install_element(CONFIG_NODE, &bgp_snmp_traps_rfc4273_cmd);
+	install_element(CONFIG_NODE, &bgp_snmp_traps_bgp4_mibv2_cmd);
+
+	SET_FLAG(bgp_snmp_traps_flags,
+		 BGP_SNMP_TRAPS_RFC4273_ENABLED);
+	SET_FLAG(bgp_snmp_traps_flags,
+		 BGP_SNMP_TRAPS_BGP4MIBV2_ENABLED);
+}
+
+int bgp_cli_snmp_traps_config_write(struct vty *vty)
+{
+	vty_out(vty, "!\n");
+	if (!CHECK_FLAG(bgp_snmp_traps_flags, BGP_SNMP_TRAPS_RFC4273_ENABLED))
+		vty_out(vty, "traps rfc4273 disabled\n");
+	if (!CHECK_FLAG(bgp_snmp_traps_flags, BGP_SNMP_TRAPS_BGP4MIBV2_ENABLED))
+		vty_out(vty, "traps bgp4-mibv2 disabled\n");
```

Comment:
Config output is completely different from DEFUN commands. `bgp snmp` is missing and `disableD` has an extra D. The latter should be changed anyway according to my previous comment.

Reply:
taken into account